week-4/D-methods/exercise-5.js

```diff
@@ -16,8 +16,29 @@ var coffeeMachine = {
     flatWhite: 3.0
   },
   insertedAmount: 0,
-  insertMoney: function(amount) {},
-  getCoffee: function(coffee) {}
+  insertMoney: function(amount) {
+    this.insertedAmount = amount;
+  },
+  getCoffee: function(coffee) {
+    if (
+      coffee === "cappuccino" &&
+      this.insertedAmount >= this.prices.cappuccino
+    ) {
+      return "Please take your " + coffee;
+    } else if (
+      coffee === "blackCoffee" &&
+      this.insertedAmount >= this.prices.blackCoffee
+    ) {
+      return "Please take your " + coffee;
+    } else if (
+      coffee === "flatWhite" &&
+      this.insertedAmount >= this.prices.flatWhite
+    ) {
+      return "Please take your " + coffee;
+    } else {
+      return "Sorry you don't have enough money for a " + coffee;
```

Comment:
I would just use a plain string in each case, rather than appending the variable `coffee`, so "Please take your cappuccino", since you know that's what it is. However this code is quite repetitive (it has `return "Please take your " + coffee;` 3 times). It might be better to separate the lookup of the price from the printing out of the "Please take your ..." messages. You could have something like this:
```
  getCoffee: function(coffee) {
    var price = this.prices[coffee];

    if (this.insertedAmount >= price) {
      return "Please take your " + coffee;
    }   
    return "Sorry you don't have enough money for a " + coffee;
  }
```

Reply:
your solution looks perfect!